usr/sbin/p11sak/p11sak.c

```diff
@@ -2892,8 +2965,14 @@ int main(int argc, char *argv[])
 
     /* Parse p11sak_defined_attrs.conf */
     rc = parse_file();
-    if (rc != CKR_OK) 
+    switch (rc) {
```

Comment:
Why not simply `if (rc == CKR_DATA_INVALID)`?  That is the only remaining exit code that leads to a failure.

Reply:
parse_file() may also return CKR_CANCEL or CKR_HOST_MEMORY

Reply:
Yeah, I would get rid of these two.  The `CKR_CANCEL` case is another example where this tool cannot be used inside sandboxes or user namespaces, and the `CKR_HOST_MEMORY` can simply be replaced by a buffer local to this function of some size (and a bailout if the path does not fit within this buffer).

Reply:
OK, now using a local buffer of PATH_MAX bytes and snprintf. CKR_CANCEL does not lead to an error anymore.

Reply:
That is standard Linux procedure.  Take bashrc as an example.  You have `/etc/bashrc` and `~/.bashrc` since you do not want to see all these user-config files per default when issuing `ls`.